### PR TITLE
Translate Rvalue::Repeat and provide a custom vec macro

### PR DESCRIPTION
**Pull request by voidc**

### CODE REVIEW

File: creusot/tests/should_succeed/syntax/13_vec_macro.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+extern crate creusot_contracts;
+
+use creusot_contracts::{vec, *};
```
Comment (voidc) [inline creusot/tests/should_succeed/syntax/13_vec_macro.rs:3]: Sadly, rustc won't let us glob import the vec macro (as it shadows the builtin macro)

File: creusot-contracts/src/std/slice.rs [hunk 2]
```diff
@@ -274,6 +277,9 @@ extern_spec! {
             forall<j:usize> i <= j && @j < (@self).len() ==> x.deep_model() < self.deep_model()[@j])]
         fn binary_search(&self, x : &T) -> Result<usize, usize>
             where T: Ord + DeepModel,  T::DeepModelTy: OrdLogic,;
+
+        #[ensures(@result == @self_)]
+        fn into_vec<A: Allocator>(self_: Box<Self, A>) -> Vec<T, A>;
```
Comment (xldenis) [inline creusot-contracts/src/std/slice.rs:282]: I think when we update to `syn 2.0` we'll be able to fix this glitch in the macros that forces us to write `self_`. previous versions of `syn` did not allow custom receiver types.

**Comment by xldenis** [comment]
lgtm! merge when ci passes.

**Comment by xldenis** [comment]
It seems like whenever we touch some specific part of `creusot-contracts` the rbt test breaks.. that sort of narrows things but not a ton..

**Pull request merged**